live/client_test.py

```diff
@@ -27,203 +34,174 @@ def mock_response(status=200, content="CONTENT",
-                                    'mediastore_container_master':
-                                        'https://hu5n3jjiyi2jev.data.media'
-                                        'store.us-east-1.amazonaws.com/master',
-                                    'mediastore_container_backup':
-                                        'https://hu5n3jjiyi2jev.data.medias'
-                                        'tore.us-east-1.amazonaws.com/backup',
-                                    'channel': "1", 'device_name': "0"})
+    client = ElementalLive(ELEMENTAL_ADDRESS, USER, API_KEY)
+    response = client.send_request(
+        'POST', f'{ELEMENTAL_ADDRESS}/live_events', HEADERS, REQUEST_BODY)
 
-    request_to_elemental = mock_request.call_args_list[0][1]
-    assert request_to_elemental['url'] == 'http://elemental' \
-                                          '.dev.cbsivideo.com/live_events'
-    assert request_to_elemental['method'] == 'POST'
-    assert request_to_elemental['headers']['Accept'] == 'application/xml'
-    assert request_to_elemental['headers']['Content-Type'] == 'application/xml'
+    assert response.text == response_from_elemental_api
     assert response.status_code == 201
 
 
 @mock.patch('requests.request')
-def test_delete_event_with_authentication(mock_request):
-    """test delete_event method"""
-    response_from_elemental_api = open(
-        'live/templates/sample_response_for_delete.xml').read()
-    mock_request.return_value = mock_response(
-        status=200, content=response_from_elemental_api)
+def test_send_request_should_raise_InvalidRequest_on_RequestException(
+        mock_request):
+    """test create_event method on invalid request"""
+    mock_request.side_effect = requests.exceptions.RequestException()
 
-    # Set match pattern
-    delete_pattern = re.compile(
-        r'http://elemental.dev.cbsivideo.com/live_events/(\d+)')
-    client = ElementalLive("http://elemental.dev.cbsivideo.com", USER, API_KEY)
+    client = ElementalLive(ELEMENTAL_ADDRESS, USER, API_KEY)
 
-    # Mock delete function
-    response = client.delete_event(event_id=42)
-
-    request_to_elemental = mock_request.call_args_list[0][1]
-    assert delete_pattern.match(request_to_elemental['url'])
-    assert request_to_elemental['method'] == 'DELETE'
-    assert response.status_code == 200
+    with pytest.raises(InvalidRequest):
+        client.send_request(
+            'POST', f'{ELEMENTAL_ADDRESS}/live_events', HEADERS, REQUEST_BODY)
 
 
 @mock.patch('requests.request')
-def test_delete_event_without_authentication(mock_request):
-    """test delete_event method"""
+def test_send_request_should_raise_InvalidResponse_on_invalid_status_code(
+        mock_request):
+    """test create_event method on invalid response"""
     response_from_elemental_api = open(
-        'live/templates/sample_response_for_delete.xml').read()
+        'live/templates/fail_to_create_response.xml').read()
```

Comment:
Probably not a big deal in these tests, but we should always close files. Easiest way to do that in python is using `with`. If you have all your test file fixtures into folder `live/templates` it could simplify the tests even more:

```python
def file_fixture(name):
    with open(os.path.join("live/templates", name)) as f:
        return f.read()
````
```python
response_from_elemental_api = file_fixture("fail_to_create_response.xml")
```